unison-cli/src/Unison/CommandLine.hs

```diff
@@ -123,7 +123,12 @@ parseInput getRoot currentPath numberedArgs patterns segments = runExceptT do
     command : args -> case Map.lookup command patterns of
       Just pat@(InputPattern {parse}) -> do
         let expandedNumbers :: [String]
-            expandedNumbers = foldMap (expandNumber numberedArgs) args
+            expandedNumbers =
+              foldMap
+                (expandNumber numberedArgs)
+                -- don't expand args of type "raw"
+                (args `zip` map ((/=) (Just "raw") . argName) [0 ..])
```

Comment:
I might back this out. At one point I was considering having `rewrite` take the rule right there on the command line, which was annoying if the rule happened to contain a numeric literal! So I introduced an argument type "raw" which is ignored by the expansion logic.

It's not currently needed by this PR. WDYT, should I just back it out?

Reply:
I think probably less confusing to just back it out.